### DISCUSSION TITLE
createStyles typescript problems

### DISCUSSION THREAD

**Discussion by 7iomka** [mantinedev/mantine#2033] [Q&A]
Hi,
I create styles for my component
```ts
import { inputSizes } from '@/config/theme/sizes';

interface FloatingInputStylesParams extends InputStylesParams {
  floating: boolean;
}

const useContainedFloatingStyles = createStyles(
  (theme, { floating, size }: FloatingInputStylesParams) => ({

    input: {
      paddingTop: theme.fn.size({ size, sizes: inputSizes }) / 4,
    },
    required: {
      transition: 'opacity 150ms ease',
      opacity: floating ? 1 : 0,
    },
  }),
);

```

Then, 

```tsx
const defaultProps: Partial<TextInputProps> = {
  size: 'md',
};

//...


const {
  inputRef,
  value: passedValue,
  onFocus,
  onBlur,
  defaultValue,
  onChange,
  ...rest
} = useComponentDefaultProps('TextInput', defaultProps, props);

// use styles
 const { classes } = useContainedFloatingStyles({
    floating: String(value).trim().length !== 0 || focused,
  });
```

But I have error here
<img width="976" alt="image" src="https://user-images.githubusercontent.com/18473137/183257802-5b5315a5-1fe5-40f5-a99f-335e81a272ef.png">

What is correct way to typing createStyles when you want to use user custom params along with all component params?
If I use `InputProps` instead of `InputStylesParams`, I also need to check all params like `size`, but this is weird, because it is always have settled with defaultValue.
Thanks.

**Comment by rtivital** [comment]
Do not extend InputStylesParam type in your styles, use only properties that you need in your styles, in your case you will need only `floating`

**Reply by 7iomka** [reply]
> Do not extend InputStylesParam type in your styles, use only properties that you need in your styles, in your case you will need only `floating`

nope, in my case I also need `size` - and with it I have a problem

I also tried: 

```
interface FloatingInputStylesParams {
  floating: boolean;
  size: MantineSize;
}

export const useContainedFloatingStyles = createStyles(
  (theme, { floating, size }: FloatingInputStylesParams) => ({
....
....

const {
  size,
  ...rest
} = useComponentDefaultProps('TextInput', defaultProps, props);


const { classes } = useContainedFloatingStyles({
  floating: String(value).trim().length !== 0 || focused,
  size: size,
});
```

But I still receive error even if size is in defaultProps
<img width="1025" alt="image" src="https://user-images.githubusercontent.com/18473137/183258367-c4e90761-bbcb-4d04-9047-7b11a5ffd0ce.png">

Why?
Of course I can hardfix with
`size: size!` but this is not a good way.

**Reply by rtivital** [reply]
Then make sure that you have size, provide default value for example

**Reply by 7iomka** [reply]
> useComponentDefaultProps

I already showed you above that I set it to default values, but it still swears, because the type of `size` from useComponentDefaultProps returns an optional value
Why?

**Reply by rtivital** [reply]
Because you have `size?:` in your component props, `useComponentDefaultProps` infers type from there – it does not know what default values you've provided. We do not have such issues in mantine codebase because we do not use strict mode.
